include/wl/j_moves.h

```diff
@@ -26,6 +26,20 @@ class JackMove {
 
 	constexpr MapNode::ID destination() const noexcept { return where_to_; }
 	constexpr std::optional<JackResource> cost() const noexcept { return resource_; }
+
+	constexpr bool operator==(const JackMove& other) const noexcept {
+		return (where_to_ == other.where_to_) &&
+		       (resource_ == other.resource_);
+	}
+	constexpr bool operator!=(const JackMove& other) const noexcept {
+		return !(*this == other);
+	}
+
+	constexpr bool operator<(const JackMove& other) const noexcept {
+		return (where_to_ < other.where_to_) ||
+			   (   (where_to_ == other.where_to_) &&
+			       (resource_ < other.resource_));
+	}
```

Comment:
This defines the result of the expression `move1 < move2` where both variables have the type `wl::JackMove`.

This is a little more interesting than defining equality, since "less than" is used to establish an **ordering** of instances of a type.

Arbitrarily, I have decided to sort first by location, and then by whether the move has a cost or not, ie:
  - if destination node number is less: **LESS**
  - else if destination node number is greater: **GREATER**
  - else (since destination node is the same): if resource is less, then **LESS**, otherwise **GREATER**

The *reason* I've added ordering for this type is for testing: we need a type to be orderable to be able to keep it in a `std::set` (a set of unique instances of a type) or `std::map` (which are associative container, with some unique instance of a type as the key and some other type as the associated value). I want to keep `wl::JackMove` in a set because:
 - I know the total set of moves that should be available to Jack in the tests, **but**
 - I don't care what order we find them in